databricks_cli/cli.py

```diff
@@ -70,5 +74,119 @@ def cli():
 cli.add_command(repos_group, name='repos')
 cli.add_command(unity_catalog_group, name='unity-catalog')
 
+
+def _trampoline_into_new_cli():
+    trampoline_disable_env_var = 'DATABRICKS_CLI_DO_NOT_EXECUTE_NEWER_VERSION'
+    if os.environ.get(trampoline_disable_env_var) is not None:
+        return
+
+    # Try to trampoline only if we're running the CLI as 'databricks'.
+    if os.path.basename(sys.argv[0]) != 'databricks':
```

Comment:
Do we want to different between running as a full path and just command name?

Reply:
I'd like to but unfortunately, this is always set to the full path if it was resolved by looking at $PATH.

We can add an additional check to see if the path is absolute as well.

Then you can still call it as `bin/databricks` or something and it doesn't show the warning.

Reply:
TBH, for consistency it's probably fine to keep the check as is. If folks intend to run it they can use the environment variable to force it to and silence the error.